lib/sth_database.js

```diff
@@ -1841,6 +1841,490 @@ function getNotificationInfo(data, callback) {
+              '_id.entityType': entityType
+            };
+            if (attrName) {
+              findCondition['_id.attrName'] = attrName;
+            }
+          } else {
+            findCondition = {
+              entityId: entityId,
+              entityType: entityType
+            };
+            if (attrName) {
+              findCondition.attrName = attrName;
+            }
+          }
+          collection.remove(
+            findCondition,
+            {
+              writeConcern: {
+                w: !isNaN(sthConfig.WRITE_CONCERN) ? parseInt(sthConfig.WRITE_CONCERN, 10) : sthConfig.WRITE_CONCERN
+              }
+            },
+            callback
+          );
+        }
+      }
+    );
+  } else {
+    var databaseName = getDatabase(service);
+    var database = db.db(databaseName);
+    var collectionName = isAggregated ?
+      getCollectionName4Aggregated(
+        {
+          databaseName: databaseName,
+          servicePath: servicePath
+        }
+      ) :
+      getCollectionName4Events(
+        {
+          databaseName: databaseName,
+          servicePath: servicePath
+        }
+      );
+
+    database.collections(function (err, collections) {
+      if (err) {
+        if (callback) {
+          return process.nextTick(callback.bind(null, err));
+        }
+      }
+
+      var dropCollectionFunctions = [];
+      collections.forEach(function (collection) {
+        if (sthConfig.SHOULD_HASH) {
+          dropCollectionFunctions.push(async.apply(dropHashedCollection, collection.collectionName, data, options));
+        } else if (collection.collectionName === collectionName) {
+          dropCollectionFunctions.push(async.apply(dropCollection, collection.collectionName, service));
+        }
+      });
+
+      async.parallel(
+        dropCollectionFunctions,
+        callback
+      );
+    });
+  }
+}
+
+/**
+ * Removes the data associated to the attributes specified in the provided data in case of collection per
+ *  entity data model
+ * @param {object} data It is an object including the following properties:
+ *  - {string} service The service
+ *  - {string} servicePath The service path
+ *  - {string} entityId The entity id
+ *  - {string} entityType The entity type
+ *  - {string} attrName The attribute name. In case no attribute name is provided, all the attributes of the
+ *    provided entity are removed
+ * @param {object} options It is an object including the following properties:
+ *  - {boolean} isAggregated Flag indicating if the operation refers to aggregated data. Raw data otherwise
+ * @param callback The callback to call with error or the result of the operation
+ */
+function removeDataIfCollection4Entity(data, options, callback) {
+  var service = data.service,
+      servicePath = data.servicePath,
+      entityId = data.entityId,
+      entityType = data.entityType,
+      attrName = data.attrName;
+  var isAggregated = options.isAggregated;
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-sth-comet/commit/7cf94a6c7d3ea3a1a0f7f869ed227f63d43f9dd1